src/pages/teachers/lessons/[lessonSlug]/downloads-auth.tsx

```diff
@@ -0,0 +1,9 @@
+/* istanbul ignore file */
+// This page should be deleted once the `use-auth` feature flag is removed
+import { withPageAuthRequired } from "@auth0/nextjs-auth0/client";
+
+import LessonDownloadsPage from "./downloads";
+
+export { getStaticPaths, getStaticProps } from "./downloads";
+
+export default withPageAuthRequired(LessonDownloadsPage);
```

Comment:
This isn't ideal. The `withPageAuthRequired` hoc will get the job done but it does the check on the client before allowing the page to render.

The original intention was to use middleware which would be much cleaner:

https://github.com/oaknational/Oak-Web-Application/blob/3c9229657661708f93a4162fbedd58eb8d2e4129/src/middleware.ts#L1-L11 

But owing to some issues getting access to OWAs environment vars in the edge runtime I changed approach to avoid getting bogged down.

Whichever auth provider we go with we will need to work to resolve this (in particular Clerk can only run in middleware it seems)

Reply:
this seems like a good comprimise! just need to make sure we prioritise resolving this as soon as we know which provider we're going with